.github/workflows/ci.yml

```diff
@@ -1,41 +1,51 @@
 name: CI
-on: [push, pull_request]
+on:
+  pull_request:
+  merge_group:
 
 jobs:
-  ci:
-    name: CI
+  test:
+    name: Test
     runs-on: ubuntu-latest
-    permissions:
-      id-token: write
     steps:
       - uses: actions/checkout@v4
 
-      - name: Run rustfmt
-        run: cargo fmt -- --check
+      - uses: Swatinem/rust-cache@v2
 
-      - name: Run clippy
-        run: cargo clippy -- -Dwarnings
+      - name: Build
+        run: cargo build
 
       - name: Run tests
         run: cargo test
 
+      - name: Run rustfmt
+        run: cargo fmt -- --check
```

Comment:
why you moved cargo fmt after tests? This check takes less than running the tests, so it's better if it fails earlier

Reply:
resolving to try to merge this.

Reply:
It doesn't really matter on this repo IMO, but I think that it's better when the more important checks fail first, rather than the faster checks. If you have a test failure, you want to know about that immediately and fix it sooner, rather than first fixing fmt, then realizing that you have a test failure, and then you might as well fix fmt again because of the changes required to fix tests :laughing: